torch/_refs/__init__.py

```diff
@@ -5744,21 +5744,52 @@ def log_normal(self, mean=1, std=2, generator=None):
 
 
 # TODO: add support for functionalization aten.normal_functional
+# NOTE: we cannot use the out wrapper because of normal.float_float_out
 @register_decomposition(aten.normal)
-@out_wrapper()
 @elementwise_type_promotion_wrapper(
     type_promoting_args=("self",),
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
-def normal(self, mean=0, std=1, generator=None):
+def normal(
+    mean=0,
+    std=1,
+    shape=None,
+    *,
+    generator=None,
+    out=None,
+    dtype=None,
+    layout=None,
+    device=None,
+    pin_memory=None,
+):
     assert generator is None
-    torch._check(std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}")
+    assert layout is None
+
+    if isinstance(std, Tensor):
+        shape = std.shape
+        dtype = std.dtype
+        device = std.device
+    else:
+        torch._check(
+            std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}"
+        )
+
+    if isinstance(mean, Tensor):
+        shape = _broadcast_shapes(shape, mean.shape)
+        dtype = mean.dtype
+        device = mean.device
+
+    if out is not None:
+        assert isinstance(out, Tensor)
+        dtype = out.dtype
+        device = out.device
```

Comment:
How would you trigger this path in eager? I don't think this is correct. The `out` implementation should be done via `@out_wrapper()` unless we have very good reasons not to do so.

Reply:
nvm, this is thje path when both are floats and the user provides the size.